gym/wrappers/gray_scale_observation.py

```diff
@@ -32,11 +32,12 @@ def __init__(self, env: gym.Env, keep_dim: bool = False):
         self.keep_dim = keep_dim
 
         assert (
-            len(env.observation_space.shape) == 3
+            isinstance(env.observation_space, Box)
+            and len(env.observation_space.shape) == 3
             and env.observation_space.shape[-1] == 3
         )
 
-        obs_shape = self.observation_space.shape[:2]
+        obs_shape = env.observation_space.shape[:2]
```

Comment:
This looks like it could change the behavior of the code, why is this necessary?

Reply:
Good point, changed back. The assert statements fixed the problem